src/txorphanage.h

```diff
@@ -21,40 +21,46 @@ extern RecursiveMutex g_cs_orphans;
 class TxOrphanage {
 public:
     /** Add a new orphan transaction */
-    bool AddTx(const CTransactionRef& tx, NodeId peer) EXCLUSIVE_LOCKS_REQUIRED(g_cs_orphans);
+    bool AddTx(const CTransactionRef& tx, NodeId peer) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex);
 
     /** Check if we already have an orphan transaction (by txid or wtxid) */
-    bool HaveTx(const GenTxid& gtxid) const LOCKS_EXCLUDED(::g_cs_orphans);
-
-    /** Get an orphan transaction and its originating peer
-     * (Transaction ref will be nullptr if not found)
+    bool HaveTx(const GenTxid& gtxid) const EXCLUSIVE_LOCKS_REQUIRED(!m_mutex);
```

Comment:
I used to be confused about the different lock granularity for these methods. Thanks for moving all the locks inside the function. 
But can we use `LOCKS_EXCLUDED(m_mutex)` instead of `EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)` for these methods?
see https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#excludes

Reply:
Using a negative annotation (`EXCLUSIVE_LOCKS_REQUIRED(!foo)`) is preferable in order to avoid "false negatives in some cases" that that link warns about (ie, the compiler not giving warnings about code that could result in double locking occurring), and `Mutex` is set up to require the use of negative annotations. That means making the change you suggest results in the error:

```
txorphanage.cpp:167:5: error: calling function 'MaybeCheckNotHeld' requires negative capability '!m_mutex' [-Werror,-Wthread-safety-analysis]
    LOCK(m_mutex);
```